src/sysfiles.c

```diff
@@ -490,6 +501,51 @@ InitInfoFunc SyLoadModule ( const Char * name )
 
 #endif
 
+/****************************************************************************
+**
+*F * * * * * * * * * * finding location of executable * * * * * * * * * * * *
+*/
+
+char GAPExecLocation[LONG_PATH_SIZE] = { 0 };
+
+void SetupGAPLocation()
+{
+    char locBuf[LONG_PATH_SIZE] = { 0 };
+    Int4 length = 0;
+    /* Some of these may fail. If so, the path will just not get set */
+#if SYS_IS_CYGWIN32
+    _get_pgmptr(GAPExecLocation);
+#elif SYS_IS_DARWIN
+    char exePath[PATH_MAX];
+    uint32_t len = sizeof(locBuf);
+    if (_NSGetExecutablePath(locBuf, &len) != 0) {
+        exePath[0] = '\0'; // buffer too small (!)
+    }
+#elif SYS_IS_LINUX
+    readlink("/proc/self/exe", locBuf, sizeof(locBuf));
```

Comment:
According to stackoverflow, this will not work on some Ubuntu systems (where one needs root privileges to read the file).